dependencies: update cgroups, ttrpc, ebpf versions

akhilerm opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?


/kind cleanup


#### What this PR does / why we need it:

- update github.com/containerd/cgroups to v1.1.0
- update github.com/containerd/ttrpc to v1.2.1
- github.com/cilium/ebpf to v0.9.1

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:
cilium/ebpf package was updated automatically while updating the cgroups and ttrpc modules

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

akhilerm commented:
/cc @dims

bart0sh commented:
@liggitt btw, I've noticed some unsuccessful attempts to enable dependabot for k/k, e.g. https://github.com/kubernetes/kubernetes/pull/113371. Any thoughts/objections about enabling it at some point?

dims commented:
/release-note-none
/approve
/lgtm

bart0sh commented:
/release-note-none

liggitt commented on vendor/github.com/cilium/ebpf/btf/marshal.go:
```diff
@@ -0,0 +1,422 @@
+func init() {
+	kernelEncoderOptions = encoderOptions{
+		ByteOrder:        internal.NativeEndian,
+		StripFuncLinkage: haveFuncLinkage() != nil,
```
this looks like it's adding unconditional setup on in an `init()` function for anything that links this package... is that new? if so, what is the cost of this, and are there any possible negative effects for binaries that don't actually use ebpf but link in this package?

akhilerm replied:
This was added as part of BTF marshalling that was added in v0.10.0. The init function was later removed in commits after the tag creation.

akhilerm replied:
Even in latest version of ebpf package, instead of `init()`, it has been moved to variables of package scope . Ref : https://github.com/cilium/ebpf/blob/master/btf/marshal.go#L20. This would still involve invoking the func.

liggitt commented on vendor/github.com/cilium/ebpf/internal/sys/signals.go:
```diff
@@ -0,0 +1,82 @@
+
+var profSet unix.Sigset_t
+
+func init() {
+	if err := sigsetAdd(&profSet, unix.SIGPROF); err != nil {
+		panic(fmt.Errorf("creating signal set: %w", err))
+	}
+}
```
same question here, this looks new and it's unclear what the impact/failure modes could be for binaries that link in this package but don't actually use ebpf at all

BenTheElder replied:
IMHO: We should reach out to upstream about moving this out of init() / var setup in favor of an explicit `Init()` or similar. I don't think it's reasonable for libraries to automatically register signal handlers regardless.

liggitt commented:
dependency update mechanics look good, a couple questions about ebpf init() functions, and would like an ack from a node approver on the content of the updates

liggitt commented:
> @liggitt btw, I've noticed some unsuccessful attempts to enable dependabot for k/k, e.g. #113371. Any thoughts/objections about enabling it at some point?

I have mixed feelings:
* we have at least some dependencies that are poorly behaved (e.g. https://github.com/kubernetes/kubernetes/issues/107499), and dependabot will push us to versions that break
* dependabot won't understand our scripts to manage dependencies for multiple modules, and will open PRs that just update go.mod files in ways that won't actually work

dims commented:
/assign @SergeyKanzhelev @mrunalp

bart0sh commented:
@liggitt
> I have mixed feelings:
> 
> * we have at least some dependencies that are poorly behaved (e.g. [remove use of github.com/imdario/mergo (currently use is incompatible with latest mergo version, latest mergo version introduces bugs) #107499](https://github.com/kubernetes/kubernetes/issues/107499)), and dependabot will push us to versions that break
> * dependabot won't understand our scripts to manage dependencies for multiple modules, and will open PRs that just update go.mod files in ways that won't actually work

What about using just as a notifier about dependency updates? I'm not familiar with our scripts, are they able to do that?

BenTheElder commented:
> What about using just as a notifier about dependency updates? 

We have such a large set of deps and carefully review changes to them. I don't think we have the bandwidth to update all the deps constantly (versus updating for known bug fixes or CVEs) and still maintain careful inspection anyhow.

If we did want to update everything all the time, there's no reason a periodic CI job couldn't be stood up to notify.

akhilerm commented:
@liggitt @BenTheElder I will split the PR into two, one for updating the cgroups,ttrpc and ebpf to 0.9.1, so the  BTF related changes that were added in 0.10.0 of ebpf can be discussed separately.

akhilerm commented:
Created https://github.com/kubernetes/kubernetes/pull/117457 with just the ebpf changes so the {cgroup, ttrpc} dependency updates can be unblocked.

akhilerm commented:
/test pull-kubernetes-node-e2e-containerd

dims commented:
leaving this for a rebase and follow ups with Ben/Jordan

/unassign

akhilerm committed:
chore: update cgroups and ttrpc versions

- update github.com/containerd/cgroups to v1.1.0
- update github.com/containerd/ttrpc to v1.2.1

Signed-off-by: Akhil Mohan <akhilerm@gmail.com>

bobbypage commented:
This looks good to me from node side, thanks @akhilerm for the bumps.

bobbypage commented:
/lgtm

liggitt commented:
/approve
